tapchannelmsg/wire_msgs.go

```diff
@@ -0,0 +1,354 @@
+package tapchannelmsg
+
+import (
+	"bytes"
+	"io"
+
+	"github.com/lightninglabs/taproot-assets/asset"
+	"github.com/lightninglabs/taproot-assets/fn"
+	"github.com/lightninglabs/taproot-assets/proof"
+	"github.com/lightningnetwork/lnd/funding"
+	"github.com/lightningnetwork/lnd/lnwire"
+	"github.com/lightningnetwork/lnd/tlv"
+)
+
+const (
+	// TxAssetInputProofType is the message type of the TxAssetInput
+	// message. The range for custom message types starts at 32768 according
+	// to BOLT-1.
+	TxAssetInputProofType lnwire.MessageType = 32769
+
+	// TxAssetOutputProofType is the message type of the TxAssetOutput
+	// message.
+	TxAssetOutputProofType lnwire.MessageType = 32770
+
+	// AssetFundingCreatedType is the message type of the
+	// AssetFundingCreated message.
+	AssetFundingCreatedType lnwire.MessageType = 32771
+
+	// AssetFundingAckType is the message type of the AssetFundingAck
+	// message.
+	AssetFundingAckType lnwire.MessageType = 32772
+)
```

Comment:
These values need to be modified such that they satisfy the following criteria:
1. They need to be specific to taproot-assets (lower probability of collision with other apps).
2. They need to avoid collision with the RFQ wire messages.

See here fore how that's done for RFQ: https://github.com/lightninglabs/taproot-assets/blob/1f580d7b9583887dc215b21f30eb3498e4b4dc56/rfqmsg/messages.go#L53-L61

The methodology for RFQ wire message types is also broken. It should be:
```
// This offset was chosen as the concatenation of the alphabetical index
// positions of the letters "t" (20), "a" (1), and "p" (16).
const TapMessageTypeBaseOffset = 20116 + 18617 + lnwire.CustomTypeStart
```

where the 18617 is new and comes from alphabet index pos of `R`, `F`, and `Q`.

Reply:
For this PR we need 
```
const TapMessageTypeBaseOffset = 20116 + <package-specific-offset> + lnwire.CustomTypeStart
```

Reply:
As @guggero mentioned on keybase:
`32768 < p2p custom message type int < 65535`

Therefore, this won't work:
```
const TapMessageTypeBaseOffset = 20116 + 18617 + lnwire.CustomTypeStart
```

Reply:
Good catch, I forgot about those numbers. I updated them to be `256` higher than the RFQ messages, that should give us enough space to add new messages there.